lib/git.ml

```diff
@@ -34,9 +34,22 @@ module Ls_remote = struct
     in
     List.fold_left ~f ~init:{ maybe_packed = None; not_packed = None } lines
 
+  let is_commit target = List.exists ~f:(fun (commit, _) -> commit = target)
+
+  let looks_like_commit ref =
+    String.length ref > 6
+    && Astring.(String.for_all Char.Ascii.is_hex_digit ref)
+
   let commit_pointed_by ~ref output_lines =
+    let warn_approx () =
+      Logs.warn (fun l ->
```

Comment:
How about making this a debug log instead?

I think there are valid use cases for pinning to a commit that's not the head of any branch or isn't tagged and users that rely on it will have to ignore these warnings. Making it a debug log would make it invisible by default.